### PR TITLE
Update/recommendations summary navigation

### PR DESCRIPTION
**Pull request by jboland88**
#### Changes proposed in this Pull Request:

* This PR adds some extra navigation controls to the recommendations flow to allow individual recommendation steps to be viewed by clicking on the recommendation name on the summary screen.
* On individual recommendation steps, if the recommended feature has been enabled, the recommendation screen now shows "Enabled" and provides a link to related settings for the feature.
* If the summary screen has already been viewed, individual recommendation steps also provide a link back to the summary screen.

#### Jetpack product discussion

pbNhbs-1iw-p2#comment-7724

#### Does this pull request change what data or activity we track or use?

Yes, this PR removes one analytics event and adds several new ones.
On the recommendations summary screen, the following analytics event is removed (it used to get recorded when clicking on the name of one of the recommendations in the summary list):
```
jetpack_recommendations_summary_learn_more_click
```

The following analytics events are added:
```
/* Recommendations summary screen */
// Recorded when clicking on the name of a recommended feature or resource with the feature slug passed as a prop
jetpack_recommendations_summary_step_name_click


/*  Individual feature recommendation */
// If a feature is already enabled, this is recorded when the "settings" button for the feature is clicked, feature slug passed as a prop
jetpack_recommended_feature_configure_click

// Recorded when clicking "View Summary" from the recommendation step, feature slug passed as a prop
jetpack_recommended_feature_back_to_summary_click


/* Individual resource recommendation */
// Recorded when clicking "View Summary" from the recommendation step, feature slug passed as a prop
jetpack_recommended_resource_back_to_summary_click

```

#### Testing instructions:

Before starting, it's helpful to clear some option values. If you are using the docker development environment, you can run these commands using `jetpack docker`.
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_step`
* Reload the site after clearing the options.

After clearing options:
* Navigate to `/wp-admin/admin.php?page=jetpack#/recommendations/site-type`
* Proceed through the recommendations flow, note that when viewing individual feature recommendations, each step provides a CTA to activate the feature or to skip it:

Screen Shot 2022-04-19 at 4 56 15 PM

* When you reach the summary screen, you should now be able to click on the names of each feature/ resource to view that step in the flow again.

Screen Shot 2022-04-19 at 5 40 42 PM

* After you have reached the summary screen and go back to view an individual step, you should have a new control below the CTA button to "View Summary". Clicking "View Summary" should return you to the summary screen.

Screen Shot 2022-04-19 at 4 11 26 PM

* If you did not activate any of the features in the flow, activate "Downtime Monitor" from the summary screen.
* After Downtime Monitor is activated, click on the feature name in the summary screen to view the individual recommendation step.
* Notice that, since the feature is active, an "Enabled" label shows on the step and the CTA link now links to the settings for the feature. Additionally, the skip link below the CTA button now reads "Next" instead of "Not Now". 

Screen Shot 2022-04-19 at 5 05 30 PM

* Test activating a few other features and confirm that the CTA buttons on the individual recommendations are linking to the proper settings controls for each feature.
* Return to the summary screen. If you do not have a "Resources" section, follow the test plan on #23600 to activate a conditional recommendation for Site Security. (TLDR: you can enable and disable the hello dolly plugin to activate the recommendation).
* Test that you can also click on the names of items in the "Resources" list to view their respective recommendation steps.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/index.jsx [hunk 1]
```diff
@@ -96,12 +119,36 @@ const FeaturePromptComponent = props => {
...
-					<a href={ nextRoute } onClick={ onDecideLaterClick }>
-						{ __( 'Not now', 'jetpack' ) }
-					</a>
+					{ featureActive ? (
+						<>
+							<div className="jp-recommendations-question__feature-enabled">
+								<div className="jp-recommendations-question__checkmark">
+									<Gridicon icon="checkmark-circle" size={ 24 } />
+								</div>
+								<span>Feature Enabled</span>
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/index.jsx:128]: This string should be translatable.

**Review by monsieur-z — CHANGES_REQUESTED** [review]
I left one minor comment in the code. Other than that, a couple of CSS issues may need to be fixed:
- The `New` badge is underlined when hovering the adjacent link
Screen Shot 2022-04-20 at 10 26 00 AM

- Some more padding between the check icon and reco name could help the reading:
Screen Shot 2022-04-20 at 10 26 36 AM

Everything behaved as expected. I've got a couple of questions/comments, though:
- The _Social Media Sharing_ settings link brought me to `wordpress.com`. I would have expected to stay in WPadmin.
- It's not obvious that the reco/resource names are clickable. I didn't realize it until I read the instructions carefully again. Maybe having them underlined in their normal state would help.

**Comment by jboland88** [comment]
Thanks for the feedback @monsieur-z !

> The Social Media Sharing settings link brought me to wordpress.com. I would have expected to stay in WPadmin.
Thanks for noting this! This was actually done intentionally based on some [feedback](https://github.com/Automattic/jetpack/pull/23362#discussion_r832994819) in PR #23362 since Calypso is where social services can be connected.

> It's not obvious that the reco/resource names are clickable. I didn't realize it until I read the instructions carefully again. Maybe having them underlined in their normal state would help.
Good thought, I'll try using an underline on those similar to other links in the same interface.

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/index.jsx (hunk 1 above)
Comment (travisw) [reply projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/index.jsx:128]: Looks like this missing translation still needs to be addressed, Josh.

**Comment by travisw** [comment]
It all works nicely. One thing that I noticed that could maybe be improved is with the VideoPress buttons. Both the summary screen and the individual recommendation step link out to an external page with the CTA text "How To", rather than an internal settings page like the others. That external page loads in the same browser tab though, unlike other external links which open in a new tab.

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/index.jsx (hunk 1 above)
Comment (jboland88) [reply projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/index.jsx:128]: Ack. I remember fixing it... maybe I reset this file at some point in the process 🙃 
Thanks for catching that!

**Comment by jboland88** [comment]
Thanks for the feedback @travisw !

> One thing that I noticed that could maybe be improved is with the VideoPress buttons. Both the summary screen and the individual recommendation step link out to an external page with the CTA text "How To", rather than an internal settings page like the others. That external page loads in the same browser tab though, unlike other external links which open in a new tab.

Ahh, good point. I'll update these setting buttons to open in a new tab and show the external link icon. Let me know what you think of this solution.

> There's also the missing translation that Kev noted.

🤦

File: projects/plugins/jetpack/_inc/client/recommendations/feature-summary/style.scss [hunk 2]
```diff
@@ -38,7 +38,10 @@
 		&:hover,
 		&:focus {
 			background: none;
-			text-decoration: underline;
+
+			.jp-recommendations-feature-summary__display-name-text {
+				box-shadow: 0 3px 0 0 currentColor
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-summary/style.scss:43]: I'm wondering if it's possible to use the `rna-link` mixin here.

**Review by monsieur-z — APPROVED** [review]
LGTM! I left a minor comment in the code, but feel free to disregard it.

**Comment by sdixon194** [comment]
Looks like there are some test failures here. Not too sure what's going on at first glance.

**Comment by sdixon194** [comment]
Nice, works well for me 👍

**Review by sdixon194 — APPROVED** [review]

**Pull request merged**